csrfguard-extensions/csrfguard-jsp-tags/src/main/java/org/owasp/csrfguard/tag/TokenValueTag.java

```diff
@@ -44,19 +44,19 @@
 	public int doStartTag() {
 		final CsrfGuard csrfGuard = CsrfGuard.getInstance();
 
-		if (csrfGuard.isTokenPerPageEnabled() && Objects.isNull(getUri())) {
-			throw new IllegalStateException("Must define 'uri' attribute when token per page is enabled");
+		if (csrfGuard.isEnabled()) {
+			final LogicalSession logicalSession = csrfGuard.getLogicalSessionExtractor().extract((HttpServletRequest) this.pageContext.getRequest());
+			final String tokenValue = Objects.nonNull(logicalSession) ? csrfGuard.getTokenService().getTokenValue(logicalSession.getKey(), getUri()) : null;
+
+			try {
+				this.pageContext.getOut().write(tokenValue);
+			} catch (final IOException e) {
+				this.pageContext.getServletContext().log(e.getLocalizedMessage(), e);
+			}
 		}
-
-		final LogicalSession logicalSession = csrfGuard.getLogicalSessionExtractor().extract((HttpServletRequest) this.pageContext.getRequest());
-		final String tokenValue = Objects.nonNull(logicalSession) ? csrfGuard.getTokenService().getTokenValue(logicalSession.getKey(), getUri()) : null;
-
-		try {
-			this.pageContext.getOut().write(tokenValue);
-		} catch (final IOException e) {
-			this.pageContext.getServletContext().log(e.getLocalizedMessage(), e);
+		else if (csrfGuard.isTokenPerPageEnabled() && Objects.isNull(getUri())) {
```

Comment:
This condition should be within the `if` statement above, executed when the solution is enabled.